.gitlab-ci.yml

```diff
@@ -0,0 +1,119 @@
+include:
+  - project: nci-gdc/gitlab-templates
+    ref: master
+    file:
+      - templates/global/full.yaml
+      - templates/deployments/salt-service.yaml
+
+.services:
+  allow_failure: true
+  stage: deploy
+  when: manual
+  parallel:
+    matrix:
+      - DEPLOY_SERVICE_LIST: [ portal, awgportal ]
+
+stages:
+  - test
+  - build
+  - release
+  - deploy
+
+variables:
+  npm_config_registry: https://nexus.osdc.io/repository/npm-all/
+  DEPLOY_SERVICE_TAG: ${CI_COMMIT_BRANCH}-${CI_COMMIT_SHORT_SHA}
+
+workflow:
+  rules:
+    - if: $CI_COMMIT_TAG
+      variables:
+        DEPLOY_SERVICE_TAG: ${CI_COMMIT_BRANCH}-${CI_COMMIT_TAG}
+    - when: always
+
+test:
+  allow_failure: true
+  stage: test
+  image: docker.osdc.io/node:13
+  before_script:
+    - npm install
+  script:
+    - npm run lint
+    - npm test
+
+build:portal-ui:
+  stage: build
+  services:
+    - docker:${DOCKER_VERSION}-dind
+  tags:
+    - dind
+  image: docker:${DOCKER_VERSION}-dind
```

Comment:
I'm not as familiar with this docker-in-a-docker name. Has the architecture changed? Will this name affect `tungsten.deploy`?

Reply:
Docker-in-docker is used for [building docker containers](https://docs.gitlab.com/ee/ci/docker/using_docker_build.html). It is isolated to this pipeline and does not require changes to tungsten.